File: client/src/main/java/commands/pcommands/PrintUniqueNumberOfRooms.java
package commands.pcommands;

import collection.CollectionFilter;
import commands.*;

import java.util.Set;

import static commands.ExecutionMode.CLIENT;

public final class PrintUniqueNumberOfRooms extends AbstractCommand {
  private final CollectionFilter collectionFilter;

  public PrintUniqueNumberOfRooms(CollectionFilter collectionManager) {
    super(CommandInfo.of("print_unique_number_of_rooms", "Print set of 'numberOfRooms' values", true, 0, CommandType.OTHER, CLIENT));
    this.collectionFilter = collectionManager;
  }

  @Override
  public ExecutionResult execute(ExecutionPayload payload) {
    Set<Long> numberOfRooms = collectionFilter.getUniqueNumberOfRooms();
    String message = numberOfRooms.toString();
    return ExecutionResult.valueOf(true, message);
  }
}

File: client-ui/domain/src/commonMain/kotlin/ru.moleus.kollector/domain/commands/pcommands/PrintUniqueNumberOfRooms.kt
package ru.moleus.kollector.domain.commands.pcommands

import commands.*
import ru.moleus.kollector.domain.collection.CollectionFilter

class PrintUniqueNumberOfRooms(private val collectionFilter: CollectionFilter) : AbstractCommand(
    CommandInfo.of(
        "print_unique_number_of_rooms",
        "Print set of 'numberOfRooms' values",
        true,
        0,
        CommandType.OTHER,
        ExecutionMode.CLIENT
    )
) {
    override fun execute(payload: ExecutionPayload): ExecutionResult {
        val numberOfRooms = collectionFilter.uniqueNumberOfRooms
        val message = numberOfRooms.toString()
        return ExecutionResult.valueOf(true, message)
    }
}